arch/arm/core/aarch32/userspace.S

```diff
@@ -366,9 +366,8 @@ SECTION_FUNC(TEXT, z_arm_do_syscall)
     /* BAD SYSCALL path */
     /* fixup stack frame on the privileged stack, adding ssf */
     mov r1, sp
-    push {r4,r5}
-    /* ssf is present in r1 (sp) */
     push {r1,lr}
+    push {r4,r5}
```

Comment:
I think the top of stack should contain r4, r5, to complete the stack frame, so , if I am not missing anything, this is exactly what you are doing here. 
I'd really appreciate this be sent as separate PR with a test case that reveals the bug

Reply:
bad system call will call 
`static uintptr_t handler_bad_syscall(uintptr_t bad_id, uintptr_t arg2,                                                                                                                                                                                  uintptr_t arg3, uintptr_t arg4,                                                                                                                                                                                    uintptr_t arg5, uintptr_t arg6,                                                                                                                                                                                    void *ssf)` the layout should be like below as commit b874b93 does.
```
 * |        ...          |
 * |---------------------- <---- SP
 * |  r4                 |
 * |----------------------- 
 * |  r5                 |
 * |----------------------- 
 * |  SSP                |
 * |----------------------- 
 * |  LR                 |
 * |----------------------- 
 * |        ...          |
```

but if no change here, the layout would be 
```
 * |        ...          |
 * |---------------------- <---- SP
 * |  SSP                |
 * |----------------------- 
 * |  LR                 |
 * |----------------------- 
 * |  r4                 |
 * |----------------------- 
 * |  r5                 |
 * |----------------------- 
 * |        ...          |
```
Ok, I will open a new issue to discuss this.